### PR TITLE
EX-445: msg 1007, 1008, 1033 harmonization and sanity checks

### PR DESCRIPTION
**Pull request by nsirola**
- Harmonize the field names of the antenna/receiver descriptor messages. (I did not find any other repos accessing these fields yet besides gnss-converters)
- Extract the maximum length of descriptor strings as define
- Add checks for the string length (counters are u8 fields but maximum valid value is 32, so there's possibility of writing over the buffer if input is malformed)

Used in gnss-converters PR https://github.com/swift-nav/gnss-converters/pull/159 which also passes unit tests.

### CODE REVIEW

File: c/include/rtcm3/messages.h [hunk 1]
```diff
@@ -176,18 +178,29 @@ typedef struct {
...
-  uint8_t antenna_serial_num_counter;
-  char antenna_serial_num[32];
-  uint8_t rcv_descriptor_counter;
-  char rcv_descriptor[32];
-  uint8_t rcv_fw_counter;
-  char rcv_fw_version[32];
-  uint8_t rcv_serial_num_counter;
-  char rcv_serial_num[32];
+  uint16_t stn_id; /* Reference Station ID DF003 uint12 12 */
+  uint8_t
```
Comment (pmiettinen) [inline c/include/rtcm3/messages.h]: This doesn't look too nice. Consider comment replacement and using manual formatting with `//clang-format off` and `//clang-format on`

File: c/src/decode.c [hunk 2]
```diff
@@ -855,38 +867,53 @@ rtcm3_rc rtcm3_decode_1033(const uint8_t buff[], rtcm_msg_1033 *msg_1033) {
...
   }
 
-  msg_1033->rcv_fw_counter = rtcm_getbitu(buff, bit, 8);
+  msg_1033->rcv_fw_version_counter = rtcm_getbitu(buff, bit, 8);
+  if (RTCM_MAX_STRING_LEN <= msg_1033->rcv_fw_version_counter) {
+    return RC_INVALID_MESSAGE;
+  }
   bit += 8;
-  for (uint8_t i = 0; i < msg_1033->rcv_fw_counter; ++i) {
+  for (uint8_t i = 0; i < msg_1033->rcv_fw_version_counter; ++i) {
```
Comment (pmiettinen) [inline c/src/decode.c]: Could do some macro nastiness for the most common patterns:

```C
#define GET_STR_LEN(TheBuff, TheIdx, TheOutput) do {                        \
  TheOutput = rtcm_getbitu(TheBuff, TheIdx, 8);                             \
  if (RTCM_MAX_STRING_LEN <= msg_1033->rcv_fw_version_counter) {            \
    return RC_INVALID_MESSAGE;                                              \
  }                                                                         \
  TheIdx += 8;
} while (false)

#define GET_STR(TheBuff, TheIdx, TheLen, TheOutput) do {                    \
  for (uint8_t i = 0; i < TheLen; ++i) {                                    \
    msg_1033->rcv_fw_version[i] = rtcm_getbitu(TheBuff, TheIdx, 8);         \
    TheIdx += 8;                                                            \
  }                                                                         \
} while (false)

GET_STR_LEN(buff, bit, msg_1033->rcv_fw_version_counter);
GET_STR(buff, bit, msg_1033->rcv_fw_version_counter, msg_1033->rcv_fw_version);
```

File: c/include/rtcm3/messages.h (hunk 1 above)
Comment (nsirola) [reply c/include/rtcm3/messages.h]: Shortened the comments for them to fit on one line

File: c/src/decode.c (hunk 2 above)
Comment (nsirola) [reply c/src/decode.c]: Done. Agreed that it's a bit nasty to `return` from within a macro, but this is otherwise much neater

**Pull request merged**